### DISCUSSION TITLE
Jump menus for Windows 11 Start Menu pinned apps

### DISCUSSION THREAD

**Discussion by stamminator** [ramensoftware/windhawk-mods#1094] [Mod ideas]
Jump lists (right-click context menu for apps pinned to the taskbar or Start Menu) have been one of my favorite features since Windows 7. Sadly, Microsoft's rewrite of the Start Menu in Windows 11 still doesn't support jump lists. Right-clicking a pinned app in the Start Menu just opens the barebones context menu that isn't much use.

![vtBYbxySsc](https://github.com/user-attachments/assets/104e38cc-9f2d-45ef-992c-656710c4cd11)

I use this feature so much that I've turned to both ExplorerPatcher and Start11 just to get it back, but between ExplorerPatcher's anti-virus woes and Start11 being ugly and clunky despite being paid software, I'd love to know if this is the kind of functionality that's capable of being added via a Windhawk mod.

I don't have much experience in Windows desktop development, but since the taskbar — which _does_ support jump lists — is XAML-based (WinUI 3?), couldn't much of that implementation be reused for the Start Menu?

![KduCHYzU27](https://github.com/user-attachments/assets/7af3cfbd-49e9-4e4e-af5a-6bce20bd832d)

Side note: after spending months of tinkering with my Windows installs and trying out so many apps and hacks to improve the experience, I can't believe I managed not to hear about Windhawk until just today, via ExplorerPatcher's release notes. Thanks for making this great software, and if I can contribute both monetarily and with dev time to show my thanks, I'll gladly do so.

**Comment by m417z** [comment]
Hi, thanks for the feedback. Unfortunately, that doesn't sound like a simple task. It can't be reused simply by calling the same function, as the start menu and the taskbar reside in different processes and load different modules. It might be possible to look at the way the taskbar does it and reimplement it, which might require a significant amount of effort, and in any case isn't going to be trivial.

**Reply by stamminator** [reply]
No worries. After further research, it looks like some preview builds of Windows 11 23H2 have start menu jump lists implemented anyway, so unless there's demand to backport it (which I don't think there is much of), it's better to just wait for the feature to roll out. I can't get preview builds on my work machine, and there's no word on when it will reach GA, so I await in agony 🥲
